acquia/blt
# Fixes #3086: Install sites from existing config.

Fixes #3086
--------

Changes proposed:
---------
For sites that store their configuration in the `config/default` directory (the default BLT configuration and the vast vast majority of sites I suspect), tell Drush to install the site from existing configuration. This is significantly faster than installing the site and then having to synchronize configuration afterwards.

Steps to replicate the issue:
----------
1. Create a new BLT project from scratch, including a VM if that's your thing.
2. Run `blt setup`
3. Run `drush cex`
4. Run `blt setup` once more (imagine that you are running this in the cloud for instance)

Observe that setup takes quite a while, since BLT first installs the site and then imports configuration changes.

Note that these steps will install Lightning. If you are using some other profile that implements hook_install, this will fail.

Steps to verify the solution:
-----------
1. Apply this patch
2. Repeat all of the steps above.
3. Verify that the behavior is exactly the same (because you haven't opted in).
4. Edit blt.yml and set cm.core.install_from_config to true.
5. Run `blt setup` again.

Observe that setup is blazingly fast by comparison, because the site is installed directly from existing config, and when the subsequent config import takes place, no changes need to get imported.


More details
----
Drupal internally refers to `config/default` as the 'sync' directory, which is why we key off of that. I'm not sure if we should make this optional or not. I'm not sure why you _wouldn't_ want to install a site from existing config, but it's a pretty new feature (as of Drupal 8.6 and Drush 9.4) so it could still be buggy, and maybe people just don't like change :man_shrugging:
